### DISCUSSION TITLE
constant height flight with z-ToF sensor

### DISCUSSION THREAD

**Discussion by kimseong-yeon** [bitcraze/discussions#895] [Q&A]
Hi, 

I'm making Aerial-ground drone for multiple indoor circumstances.
with variable motion strategy, i use only onboard sensors as you offers.

I understand crazyflie's kalman filter use z-tof sensor for state estimation 
But i found there are some problems at flying over box or obstacles because of the z-tof value.

i saw some solutions (https://www.bitcraze.io/2022/01/a-modified-ekf-for-constant-altitude-flights/)
but my drone's upper tof is not using because of other boards and batt, etc

so at first i use only downside z-tof value but it's not working properly as its video.

i'm testing it at cf2.1 and will apply at 15 cm size drone (it's so dangerous...)
is there any tips do you have?

**Comment by krichardsson** [comment]
> I understand crazyflie's kalman filter use z-tof sensor for state estimation
But i found there are some problems at flying over box or obstacles because of the z-tof value.

No, there is no support in the fimware to handle changes in the floor, it simply keeps the same distance. 

You can maybe find some inspiration in this PR https://github.com/bitcraze/crazyflie-firmware/pull/823
It implements what you are looking for, but we decided not to merge it to the main code base.

**Reply by kimseong-yeon** [reply]
thanks your reply.
it works as merge full code without upper sensor working.
